tiledb/sm/subarray/subarray.cc

```diff
@@ -2171,26 +2258,35 @@ Status Subarray::compute_relevant_fragment_tile_overlap(
     FragmentMetadata* meta,
     unsigned frag_idx,
     bool dense,
-    uint64_t range_num,
     ThreadPool* const compute_tp,
-    std::vector<std::vector<TileOverlap>>* const tile_overlap) {
-  auto num_threads = compute_tp->concurrency_level();
-  auto ranges_per_thread = (uint64_t)ceil((double)range_num / num_threads);
-  auto statuses = parallel_for(compute_tp, 0, num_threads, [&](uint64_t t) {
-    auto r_start = t * ranges_per_thread;
-    auto r_end = std::min((t + 1) * ranges_per_thread - 1, range_num - 1);
-    for (uint64_t r = r_start; r <= r_end; ++r) {
-      if (dense) {  // Dense fragment
-        (*tile_overlap)[frag_idx][r] = get_tile_overlap(r, frag_idx);
-      } else {  // Sparse fragment
-        const auto& range = this->ndrange(r);
-        RETURN_NOT_OK(
-            meta->get_tile_overlap(range, &((*tile_overlap)[frag_idx][r])));
-      }
-    }
+    SubarrayTileOverlap* const tile_overlap,
+    ComputeRelevantTileOverlapCtx* const fn_ctx) {
```

Comment:
@stavrospapadopoulos 

This routine (`compute_relevant_fragment_tile_overlap`) now computes only on the new ranges from the last call. For example:
```
compute_relevant_fragments(0, 10);
compute_relevant_fragment_tile_overlap(0, 10); // computes ranges [0, 10)
compute_relevant_fragments(0, 20); // might add new relevant fragments
compute_relevant_fragment_tile_overlap(0, 20); // computes ranges [10, 20)
```

I am pointing this out because if new relevant fragments have been found, this will not recompute the earlier ranges with the new relevant fragments. I have reasoned that this is OK because if there was a chance of overlap, those fragments would have been relevant for the earlier ranges.

Reply:
Yes, that is correct.